packages/cmr-client/src/EarthdataLogin.ts

```diff
@@ -0,0 +1,211 @@
+// @ts-nocheck
+import get from 'lodash/get';
+import got from 'got';
+import Response from 'got';
+
+type EarthdataGetTokenResponse = Response<{
+  body: {
+    access_token?: string,
+    token_type?: string,
+    expiration_date?: string
+  }
+}>;
+
+type EarthdataPostTokenResponse = Response<{
+  body: {
+    access_token?: string,
+    expiration_date?: string
+  }
+}>;
+
+export interface EarthdataLoginParams {
+  username: string,
+  password: string,
+  edlEnv: string,
+}
+
+/**
+ * A class to simplify requests for the Earthdata Login token used for CMR
+ *
+ * @typicalname earthdataLogin
+ *
+ * @example
+ * const { EarthdataLogin } = require('@cumulus/cmr-client');
+ *
+ * const earthdataLogin = new EarthdataLogin({
+ *  "username": "my-username",
+ *  "password": "my-password",
+ *  "edlEnv": "my-cmr-environment"
```

Comment:
```suggestion
 *  "edlEnv": "my-edl-environment"
```

Reply:
Please update all 'CMR' to 'EDL', they are two different applications

Reply:
Add unit tests for EathdataLogin class.

Reply:
Add a test file test-EarthdataLogin.js and test each method in the class, tests should cover as much code as possible, e.g.
test('retrieveToken returns a valid token')
test('retrieveToken returns undefined when there is no valid token')
test('retrieveToken throws exception where invalid user credential')